lib/agent/redirect.js

```diff
@@ -143,15 +143,17 @@ function shouldRemoveHeader (header, removeContent) {
 function cleanRequestHeaders (headers, removeContent) {
   const ret = []
   if (Array.isArray(headers)) {
-    for (let i = 0; i < headers.length; i += 2) {
-      if (!shouldRemoveHeader(headers[i], removeContent)) {
-        ret.push(headers[i], headers[i + 1])
+    const length = headers.length
+    for (let i = 0; i < length; i += 2) {
+      const header = headers[i]
+      if (!shouldRemoveHeader(header, removeContent)) {
+        ret.push(header, headers[i + 1])
       }
     }
   } else if (headers && typeof headers === 'object') {
-    for (const [key, val] of Object.entries(headers)) {
+    for (const key in headers) {
```

Comment:
please revert this change. Using `in` makes it vulnerable to prototype pollution attacks.

Reply:
But the object is given by the user. I think it should be their responsibility to prevent prototype pollution.

Reply:
Alternatively `hasOwnProperty` could be used. Or `Object.keys` instead of `Object.entries`.

Reply:
Object.keys is the better option?